aidial_adapter_openai/utils/parsers.py

```diff
@@ -52,32 +52,45 @@ def get_client(self, params: OpenAIParams) -> AsyncOpenAI:
         )
 
 
-class EndpointParser(BaseModel):
-    name: str
-
-    def parse(self, endpoint: str) -> AzureOpenAIEndpoint | OpenAIEndpoint:
-        match = re.search(
-            f"(.+?)/openai/deployments/(.+?)/{self.name}", endpoint
+def _parse_endpoint(
+    name, endpoint
+) -> AzureOpenAIEndpoint | OpenAIEndpoint | None:
+    if azure_match := re.search(
+        f"(.+?)/openai/deployments/(.+?)/{name}", endpoint
+    ):
+        return AzureOpenAIEndpoint(
+            azure_endpoint=azure_match[1],
+            azure_deployment=azure_match[2],
         )
 
-        if match:
-            return AzureOpenAIEndpoint(
-                azure_endpoint=match[1],
-                azure_deployment=match[2],
-            )
+    if openai_match := re.search(f"(.+?)/{name}", endpoint):
+        return OpenAIEndpoint(base_url=openai_match[1])
```

Comment:
Python returns None by default when there are no return statements and thus no return value.

However, here it's a slightly different situation - we do have have a return value which may be `None`.

So it would be great to see the final `return None`, even though it works as is just fine.

Reply:
Fixed